documentation/high-level-introduction.md

```diff
@@ -0,0 +1,251 @@
+
+Documentation on msbuild is both widespread and yet much of it is difficult to parse for those coming up to speed. There are also no good "high level" docs that paint a picture on how the parts are put together.
+
+The Xamarin.Android folks have written up a best [practices guide](https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/MSBuildBestPractices.md), which is more of a "graduate level" guide. 
+
+This document aims to sit somewhere in the middle, “101 college level”. It should contain more than 5 minutes of content but for those coming to msbuild with very little background in how it works.
+
+### What is msbuild?
+
+msbuild is a build system commonly used to build .NET projects. It is made up the msbuild toolchain itself, and some number of additional files provided by the SDKs which provide platform specific instructions on how to process projects of those types.
+
+msbuild is built up of a few very generic ideas:
+
+- Run a sequence of defined steps based upon input / output files and dependency ordering
+- Store single and groups of values in a global table
+- Steps can load C# library tasks and execute them
+- Include other files in the file we are processing 
+
+Built on top of those bare bones are a huge number of conventions:
+
+- Separating your data and step definitions in different files (see file structure below)
+- Creating common locations to import “SDK” files from
+- Standard variables to add source files \ references \ resources
+- Prefixing “private” tasks with underscore
+- Splitting language specific items from a Common file to CSharp and FSharp versions of the same file. 
+
+None of these are enforced by msbuild itself. The files commonly included by user projects just assume you will follow these conventions.
+
+
+## File Structure Convention
+
+Before we get into the specifics of how things work, let’s talk about the general file structure, which is built upon many standard conventions. We’ll cover the how [data is represented](#data) and the [nodes available](#XML-Structure) in each shortly. 
+
+### Text Files
+
+Instead of including every single detail of a build in the user projects, this information tends to be split into multiple places. Some of these are provided by SDKs are stored in standard global locations, for example: `/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/` 
+
+
+- Project files (csproj)
+  - These are user provided files describing the files, references, and settings for the build in question. 
+  - These tend to contain many Reference items and additions to the Compile and None item group
+  - These tend to include one single project specific “target” file that include the rest of the files needed to define the entire build.
+`<Import Project="$(MSBuildExtensionsPath)\Xamarin\Mac\Xamarin.Mac.CSharp.targets" />`
+  - Solution files (sln) just combine one or more projects for editing and building. The project file is main level of build knowledge.
```

Comment:
I would also specify here that solution files specify build dependencies and configurations (i.e. this is mac only)